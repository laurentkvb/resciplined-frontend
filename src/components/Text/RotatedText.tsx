import styled from "styled-components";
import { color, flex, layout, position, space, typography } from "styled-system";
import { SystemProps } from "@models/styledSystem";
import React from "react";

const RotatedText : React.FC<SystemProps> = styled.div<SystemProps>`

    transform: rotate(90deg);
    z-index: 100;
    font-weight: bold;
    width: 200px;
    font-size: 30px;
    
   
    &&& {
    ${color}
    ${layout}
    ${position}
    ${space}
    ${flex}
    ${typography}
  }
`;

export default RotatedText;
