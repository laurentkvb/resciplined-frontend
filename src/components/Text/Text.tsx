import React, { ReactNode } from "react";

import { TextComponent, TextProps } from "./text.styles";

interface Props extends TextProps {
  children?: ReactNode;
}

const Text = (props: Props) => <TextComponent data-testid="Text" {...props} />;

export default Text;
