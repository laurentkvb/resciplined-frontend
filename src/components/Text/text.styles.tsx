import styled from "styled-components";
import { variant, color, typography, space, layout } from "styled-system";
import { SystemProps } from "@models/styledSystem";
import React from "react";


export interface TextProps extends SystemProps {
  variant?: string | string[];
}

export const TextComponent: React.FC<TextProps> = styled.span<TextProps>`
  ${variant({
    variants: {
      h1: {
        display: "block",
        fontFamily: "Andada",
        fontSize: ["0px", "70px", "70px", "85px", "110px", "110px", "110px"],
        lineHeight: ["0px", "63px", "63px", "63px", "94px", "94px", "94px"],
        fontWeight: 600,
        textTransform: "uppercase"
      },
      h2: {
        display: "block",
        fontFamily: "Andada",
        fontSize: ["0px", "60px", "60px", "60px", "80px", "80px", "80px"],
        lineHeight: ["0px", "50px", "50px", "50px", "70px", "70px", "70px"],
        fontWeight: 600,
        textTransform: "uppercase"
      },
      h3: {
        display: "block",
        fontFamily: "Crimson",
        fontSize: ["0px", "50px", "50px", "50px", "60px", "60px", "60px"],
        lineHeight: ["0px", "50px", "50px", "50px", "56px", "56px", "56px"],
        fontWeight: 800,
      },
      bodyL: {
        display: "inline",
        fontFamily: "Crimson",
        fontSize: ["0px", "22px", "22px", "22px", "28px", "28px", "28px"],
        lineHeight: ["0px", "28px", "28px", "28px", "37px", "37px", "37px"],
        fontWeight: 300,
      },
      bodyM: {
        display: "inline",
        fontFamily: "Crimson",
        fontSize: ["0px", "18px", "18px", "18px", "24px", "24px", "24px"],
        lineHeight: ["0px", "24px", "24px", "24px", "29px", "29px", "29px"],
        fontWeight: 300,
      },
      bodyS: {
        display: "inline",
        fontFamily: "Crimson",
        fontSize: ["0px", "16px", "16px", "16px", "18px", "18px", "18px"],
        lineHeight: ["0px", "20px", "20px", "20px", "24px", "24px", "24px"],
        fontWeight: 300,
      },
    }
  })}
  &&& {
    ${color}
    ${typography}
    ${space}
    ${layout}
  }
`;
