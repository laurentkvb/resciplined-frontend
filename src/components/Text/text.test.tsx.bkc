import React from "react";
import { render } from "tests/test-utils";

import Text from "./Text";

test("Text snapshot Test", () => {
  const { container } = render(
    <>
      <Text variant="title">Title</Text>
      <Text variant="body">Title</Text>
      <Text variant="paragraph">Title</Text>
    </>
  );

  // Asserts Text component to match snapshot.
  expect(container.firstChild).toMatchSnapshot();
});

test("Text", () => {
  const { getByTestId } = render(
    <Text variant="title">Lorem ipsum dolor sit amet consectetur.</Text>
  );

  const testComponent = getByTestId("Text");

  // Asserts SubTitle component renders children correctly.
  expect(testComponent.firstChild?.textContent).toBe(
    "Lorem ipsum dolor sit amet consectetur."
  );
});
