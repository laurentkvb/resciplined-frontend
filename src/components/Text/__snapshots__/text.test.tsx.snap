// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text snapshot Test 1`] = `
.c0 {
  display: block;
  font-size: 1.3125rem;
  line-height: 1.1;
  font-weight: 600;
}

@media screen and (min-width:640px) {
  .c0 {
    font-size: 1.5rem;
  }
}

<span
  class="c0"
  data-testid="Text"
>
  Title
</span>
`;
